### PR TITLE
#839 - update schema, types, and seed data

### LINKED ISSUE #839: Work Package Stages - Modify Prisma Schema
### Description

read the epic for background on this feature: #345 

### Acceptance Criteria

modify the prisma schema to have:
- a new enum `WORK_PACKAGE_STAGE` with the values specified in the epic
- this should be an optional parameter on the work package model

### Proposed Solution

1. add it to the schema
2. add it to the shared type
3. add it to the transformer and query args as needed
4. modify the seed data
5. create the [prisma migration](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8749072/Software+Prisma+FAQ#yarn-prisma%3Amigrate)

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by RChandler234**
## Changes

Updated schema, shared types, and seed data

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #839  (issue #)

### CODE REVIEW

File: src/backend/src/prisma/schema.prisma [hunk 1]
```diff
@@ -62,6 +62,13 @@ enum Scope_CR_Why_Type {
   OTHER
 }
 
+enum WP_Stage {
```
Comment (anthonybernardi) [inline src/backend/src/prisma/schema.prisma:65]: Work_Package_Stage

File: src/shared/src/types/work-package-types.ts [hunk 2]
```diff
@@ -9,3 +9,10 @@ export enum TimelineStatus {
   Behind = 'BEHIND',
   VeryBehind = 'VERY_BEHIND'
 }
+
+export enum WPStage {
```
Comment (anthonybernardi) [inline src/shared/src/types/work-package-types.ts:13]: WorkPackageStage

File: src/backend/src/transformers/projects.transformer.ts [hunk 3]
```diff
@@ -90,7 +91,8 @@ const projectTransformer = (project: Prisma.ProjectGetPayload<typeof projectQuer
         dependencies: workPackage.dependencies.map(wbsNumOf),
         expectedActivities: workPackage.expectedActivities.map(descBulletConverter),
         deliverables: workPackage.deliverables.map(descBulletConverter),
-        projectName: wbsElement.name
+        projectName: wbsElement.name,
+        stage: (workPackage.stage as WPStage) ?? undefined
```
Comment (anthonybernardi) [inline src/backend/src/transformers/projects.transformer.ts:95]: does this work? like it'll be (null as WPStage) and would that cause an error?

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
noice

File: src/backend/src/prisma/seed-data/work-packages.seed.ts [hunk 4]
```diff
@@ -3,10 +3,11 @@
  * See the LICENSE file in the repository root folder for details.
  */
 
-import { WBS_Element_Status } from '@prisma/client';
+import { WBS_Element_Status, Work_Package_Stage } from '@prisma/client';
 
 const dbSeedWorkPackage1: any = {
   projectId: 1,
+  stage: Work_Package_Stage.RESEARCH,
```
Comment (anthonybernardi) [inline src/backend/src/prisma/seed-data/work-packages.seed.ts:10]: these need to be in work package fields (the way this is set up is nonsensical tho lmao)

File: src/backend/src/transformers/work-packages.transformer.ts [hunk 5]
```diff
@@ -36,7 +36,8 @@ const workPackageTransformer = (wpInput: Prisma.Work_PackageGetPayload<typeof wo
       detail: change.detail,
       dateImplemented: change.dateImplemented
     })),
-    projectName: wpInput.project.wbsElement.name
+    projectName: wpInput.project.wbsElement.name,
+    stage: wpInput.stage
```
Comment (anthonybernardi) [inline src/backend/src/transformers/work-packages.transformer.ts:40]: should thsi be the same as in the projects transformer where it can be null?

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
awesome, just tiny stuff

File: src/backend/src/prisma/seed-data/work-packages.seed.ts (hunk 4 above)
Comment (RChandler234) [reply src/backend/src/prisma/seed-data/work-packages.seed.ts:10]: Is there any particular reason it has to go there?

Comment (anthonybernardi) [reply src/backend/src/prisma/seed-data/work-packages.seed.ts]: go look at how the seed data gets used in seed.ts -- it's very weird, especially when we could be using the built in prisma create types

**Review by anthonybernardi — APPROVED** [review]
NICE

**Pull request merged**